domains/client/domain-operator/src/domain_block_processor.rs

```diff
@@ -340,7 +341,12 @@ where
 
         let runtime_api = self.client.runtime_api();
 
-        let mut roots = runtime_api.intermediate_roots(header_hash)?;
+        // TODO: consider changing this API to return Blake3Hash
```

Comment:
Do we want to make this runtime API type change as well? In this PR, or in another one?

Reply:
This API has nothing to do with Blake3, it just happens to be the same size. State root, block hash and other things like that in the runtime are Blake2 hashes. You should not convert it to `Blake3Hash` at all.

Reply:
Yeah, I was wondering about that. Should the hash type be `H256`, or do we have a specific blake 2 hash type?

Reply:
I simply wouldn't touch it here. The hash comes from Substrate and in some cases we do use H256, but in many places it is simply a generic type. Maybe @NingLin-P has better context for this particular runtime API.

Reply:
The return value of `intermediate_roots` is Blake2 hashes but present as `Vec<[u8; 32]>` I think it is unnecessary to convert them to `H256` or specific Blake2 hash type because they are converting to `Vec<<Block as BlockT>::Hash>` in the next few lines.